### PR TITLE
fix: Fix logging of add-in discovery

### PR DESCRIPTION
**Pull request by dr1rrb**
linked https://github.com/unoplatform/uno.hotdesign/issues/2986#top

## Bugfix
Fix logging of add-in discovery

## What is the current behavior?
Logging uses  `NullLoggger`

## What is the new behavior?
Use temp console log during statup of dev-server

## PR Checklist
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

### CODE REVIEW

File: src/Uno.UI.RemoteControl.Host/Extensibility/AddIns.cs [hunk 1]
```diff
@@ -17,29 +17,54 @@ public class AddIns
 	public static IImmutableList<string> Discover(string solutionFile)
```
Comment (agneszitte) [inline src/Uno.UI.RemoteControl.Host/Extensibility/AddIns.cs:17]: can we add unit tests for these changes? for scenarios like empty frameworks, invalid solution files, command execution failures, for example

File: src/Uno.UI.RemoteControl.Host/Extensibility/AddIns.cs [hunk 2]
```diff
@@ -17,29 +17,54 @@ public class AddIns
...
 	{
 		var tmp = Path.GetTempFileName();
-		var result = ProcessHelper.RunProcess("dotnet", $"build \"{solutionFile}\" --target:UnoDumpTargetFrameworks \"-p:UnoDumpTargetFrameworksTargetFile={tmp}\" --verbosity quiet");
+		var command = $"build \"{solutionFile}\" --target:UnoDumpTargetFrameworks \"-p:UnoDumpTargetFrameworksTargetFile={tmp}\" --verbosity quiet";
```
Comment (agneszitte) [inline src/Uno.UI.RemoteControl.Host/Extensibility/AddIns.cs:20]: Do we need to add validation or logging before executing it?

File: src/Uno.UI.RemoteControl.Host/Extensibility/AddIns.cs [hunk 3]
```diff
@@ -17,29 +17,54 @@ public class AddIns
...
+		{
+			_log.Log(LogLevel.Debug, $"Found target frameworks for solution '{solutionFile}': {string.Join(", ", targetFrameworks)}.");
+		}
+
 
 		foreach (var targetFramework in targetFrameworks)
 		{
 			tmp = Path.GetTempFileName();
-			result = ProcessHelper.RunProcess("dotnet", $"build \"{solutionFile}\" --target:UnoDumpRemoteControlAddIns \"-p:UnoDumpRemoteControlAddInsTargetFile={tmp}\" --verbosity quiet --framework \"{targetFramework}\" -nowarn:MSB4057");
+			command = $"build \"{solutionFile}\" --target:UnoDumpRemoteControlAddIns \"-p:UnoDumpRemoteControlAddInsTargetFile={tmp}\" --verbosity quiet --framework \"{targetFramework}\" -nowarn:MSB4057";
```
Comment (agneszitte) [inline src/Uno.UI.RemoteControl.Host/Extensibility/AddIns.cs:53]: Do we need to add validation or logging before executing it?

File: src/Uno.UI.RemoteControl.Host/Extensibility/AddIns.cs [hunk 4]
```diff
@@ -17,29 +17,54 @@ public class AddIns
...
 			return ImmutableArray<string>.Empty;
 		}
 
+		if (_log.IsEnabled(LogLevel.Debug))
+		{
+			_log.Log(LogLevel.Debug, $"Found target frameworks for solution '{solutionFile}': {string.Join(", ", targetFrameworks)}.");
+		}
+
 
 		foreach (var targetFramework in targetFrameworks)
```
Comment (agneszitte) [inline src/Uno.UI.RemoteControl.Host/Extensibility/AddIns.cs:50]: is this quick on the main thread? or cancellation tokens needed?

File: src/Uno.UI.RemoteControl.Host/Program.cs [hunk 5]
```diff
@@ -81,14 +87,23 @@ static async Task Main(string[] args)
 					services.AddSingleton<IIdeChannel, IdeChannelServer>();
 				});
 
+			Debugger.Launch();
```
Comment (agneszitte) [inline src/Uno.UI.RemoteControl.Host/Program.cs:90]:
```suggestion
#if DEBUG
			Debugger.Launch();
#endif
```

File: src/Uno.UI.RemoteControl.Host/Program.cs [hunk 6]
```diff
@@ -81,14 +87,23 @@ static async Task Main(string[] args)
...
+
 			if (solution is not null)
 			{
 				// For backward compatibility, we allow to not have a solution file specified.
 				builder.ConfigureAddIns(solution);
 			}
+			else
+			{
+				typeof(Program).Log().Log(LogLevel.Warning, "No solution file specified, add-ins will not be loaded.");
```
Comment (agneszitte) [inline src/Uno.UI.RemoteControl.Host/Program.cs:99]: should we add additional context in the log message (e.g. potential impacts of not having a solution file) to inform the user ?

File: src/Uno.UI.RemoteControl.Host/Program.cs [hunk 7]
```diff
@@ -61,6 +62,11 @@ static async Task Main(string[] args)
 				throw new ArgumentException($"The httpPort parameter is required.");
 			}
```
Comment (agneszitte) [inline src/Uno.UI.RemoteControl.Host/Program.cs:64]: unit tests? when the solution is null, outputs correct messages for missing or invalid inputs, initialization of the LoggerFactory, for example

File: src/Uno.UI.RemoteControl.Host/Program.cs (hunk 5 above)
Comment (agneszitte) [reply src/Uno.UI.RemoteControl.Host/Program.cs:90]: or
```suggestion
#if DEBUG
			if (Debugger.IsAttached == false)
			{
				Debugger.Launch();
			}
#endif
```

**Review by agneszitte — APPROVED** [review]

File: src/Uno.UI.RemoteControl.Host/Program.cs (hunk 5 above)
Comment (dr1rrb) [reply src/Uno.UI.RemoteControl.Host/Program.cs:90]: Actually : 
```suggestion
```

File: src/Uno.UI.RemoteControl.Host/Extensibility/AddIns.cs (hunk 1 above)
Comment (dr1rrb) [reply src/Uno.UI.RemoteControl.Host/Extensibility/AddIns.cs:17]: We don't have testing yet for dev-server. Will need it at some point, but not part of this PR

File: src/Uno.UI.RemoteControl.Host/Extensibility/AddIns.cs (hunk 2 above)
Comment (dr1rrb) [reply src/Uno.UI.RemoteControl.Host/Extensibility/AddIns.cs:20]: We log the output in case of failure. This is enough for now.

File: src/Uno.UI.RemoteControl.Host/Extensibility/AddIns.cs (hunk 4 above)
Comment (dr1rrb) [reply src/Uno.UI.RemoteControl.Host/Extensibility/AddIns.cs:50]: No it's no quick, but we cannot have CT since it's at the startup of the dev-server: it has to be sync (the thread will be killed anyway)

File: src/Uno.UI.RemoteControl.Host/Program.cs [hunk 8]
```diff
@@ -81,14 +87,23 @@ static async Task Main(string[] args)
 					services.AddSingleton<IIdeChannel, IdeChannelServer>();
 				});
 
+			Debugger.Launch();
+
```
Comment (dr1rrb) [inline src/Uno.UI.RemoteControl.Host/Program.cs:91]:
```suggestion
```

File: src/Uno.UI.RemoteControl.Host/Program.cs (hunk 7 above)
Comment (dr1rrb) [reply src/Uno.UI.RemoteControl.Host/Program.cs:64]: We don't have testing yet for dev-server.

File: src/Uno.UI.RemoteControl.Host/Program.cs (hunk 6 above)
Comment (dr1rrb) [reply src/Uno.UI.RemoteControl.Host/Program.cs:99]:
```suggestion
				typeof(Program).Log().Log(LogLevel.Warning, "No solution file specified, add-ins will not be loaded which means that you won't be able to use any of the uno-studio features. Usually this indicates that your version of uno's IDE extension is too old.");
```

**Review by jeromelaban — APPROVED** [review]

**Comment by jeromelaban** [comment]
@mergifyio backport release/stable/5.5
